setup.py

```diff
@@ -56,10 +56,7 @@
               'raven==6.9.0',
               'requests==2.19.1',
               'requests-toolbelt==0.8.0',
-              # since 0.15.52, it returns ordereddict instead of pure dicts, making format_dict unhappy
-              'ruamel.yaml==0.15.51',
-              # newer versions bring way too many incompabilities, needs deeper inspection
-              # 'ruamel.yaml==0.15.34',
+              'ruamel.yaml==0.16.12',
```

Comment:
Have you checked what you get when you `load_yaml` and try to print it with `format_dict`? `ruamel.yaml` started to use `OrderedDict` instead of plain mappings, killing any readability. Hence the comment...

Reply:
not yet

Reply:
@happz hmm, seems good to me?

```python
> from gluetool.utils import load_yaml
> from gluetool.log import format_dict
> load_yaml('generate-docs.yml')
[ordereddict([('hosts', 'localhost'), ('vars', ordereddict([('sphinx_builder', 'html'), ('generate_dir', './docs'), ('log_path', '{{ generate_dir }}/build/sphinx_build_stderr.log'), ('clean', False), ('static_rst', ['source/framework.rst', 'source/modules.rst', 'source/howto-tests.rst', 'source/howto-docs.rst', 'source/index.rst', 'source/DEVELOPMENT.rst']), ('base_files', ['source/conf.py']), ('clean_files', ['build', 'source/gluetool.*'])])), ('tasks', [ordereddict([('name', 'Ensure directory hierarchy exists'), ('file', ordereddict([('path', '{{ item }}'), ('state', 'directory')])), ('with_items', ['{{ generate_dir }}/', '{{ generate_dir }}/source/'])]), ordereddict([('name', "Clean doc in '{{ generate_dir }}'"), ('shell', 'rm -rf {{ generate_dir }}/{{ item }}'), ('args', ordereddict([('removes', '{{ generate_dir }}/{{ item }}')])), ('when', 'clean'), ('with_items', '{{ clean_files }}'), ('tags', ['clean'])]), ordereddict([('name', 'Copy static_rst files'), ('copy', ordereddict([('src', 'docs/{{ item }}'), ('dest', '{{ generate_dir }}/{{ item }}')])), ('with_items', '{{ static_rst }}')]), ordereddict([('name', 'Copy base files'), ('copy', ordereddict([('src', 'docs/{{ item }}'), ('dest', '{{ generate_dir }}/{{ item }}')])), ('with_items', '{{ base_files }}')]), ordereddict([('name', 'Generate apidoc'), ('command', 'sphinx-apidoc -T -e -o "{{ generate_dir }}/source/" gluetool/')]), ordereddict([('name', 'Generate documentation'), ('command', 'sphinx-build -M "{{ sphinx_builder }}" source build'), ('register', 'sphinx_build'), ('args', ordereddict([('chdir', '{{ generate_dir }}')]))]), ordereddict([('name', 'Create error log file {{log_path}}'), ('copy', ordereddict([('content', '{{ sphinx_build.stderr }}'), ('dest', '{{ log_path }}')]))]), ordereddict([('name', 'Check log file for errors'), ('command', 'awk \'/ERROR:/ { print "Errors were detected, check log file and your docs"; exit 1; }\' "{{ log_path }}"')])])])]
> print(format_dict(load_yaml('generate-docs.yml')))
[
    {
        "hosts": "localhost",
        "tasks": [
            {
                "file": {
                    "path": "{{ item }}",
                    "state": "directory"
                },
                "name": "Ensure directory hierarchy exists",
                "with_items": [
                    "{{ generate_dir }}/",
                    "{{ generate_dir }}/source/"
                ]
            },
            {
                "args": {
                    "removes": "{{ generate_dir }}/{{ item }}"
                },
                "name": "Clean doc in '{{ generate_dir }}'",
                "shell": "rm -rf {{ generate_dir }}/{{ item }}",
                "tags": [
                    "clean"
                ],
                "when": "clean",
                "with_items": "{{ clean_files }}"
            },
            {
                "copy": {
                    "dest": "{{ generate_dir }}/{{ item }}",
                    "src": "docs/{{ item }}"
                },
                "name": "Copy static_rst files",
                "with_items": "{{ static_rst }}"
            },
            {
                "copy": {
                    "dest": "{{ generate_dir }}/{{ item }}",
                    "src": "docs/{{ item }}"
                },
                "name": "Copy base files",
                "with_items": "{{ base_files }}"
            },
            {
                "command": "sphinx-apidoc -T -e -o \"{{ generate_dir }}/source/\" gluetool/",
                "name": "Generate apidoc"
            },
            {
                "args": {
                    "chdir": "{{ generate_dir }}"
                },
                "command": "sphinx-build -M \"{{ sphinx_builder }}\" source build",
                "name": "Generate documentation",
                "register": "sphinx_build"
            },
            {
                "copy": {
                    "content": "{{ sphinx_build.stderr }}",
                    "dest": "{{ log_path }}"
                },
                "name": "Create error log file {{log_path}}"
            },
            {
                "command": "awk '/ERROR:/ { print \"Errors were detected, check log file and your docs\"; exit 1; }' \"{{ log_path }}\"",
                "name": "Check log file for errors"
            }
        ],
        "vars": {
            "base_files": [
                "source/conf.py"
            ],
            "clean": false,
            "clean_files": [
                "build",
                "source/gluetool.*"
            ],
            "generate_dir": "./docs",
            "log_path": "{{ generate_dir }}/build/sphinx_build_stderr.log",
            "sphinx_builder": "html",
            "static_rst": [
                "source/framework.rst",
                "source/modules.rst",
                "source/howto-tests.rst",
                "source/howto-docs.rst",
                "source/index.rst",
                "source/DEVELOPMENT.rst"
            ]
        }
    }
]
```

Reply:
Looks much better than it used too. When I tested it, the output looked like what you got from `load_yaml()`.